Upgrade jotai to latest `1.5.*`, fix breaking changes

Jotai 1.4 has breaking changes that come with some of its improvements for TypeScript. 

<details>
<summary>Type errors</summary>

```bash
yarn run v1.22.11
$ tsc --noEmit
src/pages/transaction-signing/components/post-conditions/fungible-post-condition.tsx(46,18): error TS2571: Object is of type 'unknown'.
src/pages/transaction-signing/components/post-conditions/fungible-post-condition.tsx(46,57): error TS2571: Object is of type 'unknown'.
src/pages/transaction-signing/components/post-conditions/fungible-post-condition.tsx(47,18): error TS2571: Object is of type 'unknown'.
src/pages/transaction-signing/components/post-conditions/fungible-post-condition.tsx(69,13): error TS2571: Object is of type 'unknown'.
src/store/accounts/account-activity.ts(26,5): error TS2554: Expected 3 arguments, but got 2.
src/store/accounts/index.ts(73,41): error TS2554: Expected 3 arguments, but got 2.
src/store/accounts/nonce.ts(16,13): error TS2554: Expected 3 arguments, but got 2.
src/store/assets/fungible-tokens.hooks.ts(7,23): error TS2345: Argument of type 'FungibleTokenMetadata | { error: string; } | undefined' is not assignable to parameter of type 'Atom<unknown>'.
  Type 'undefined' is not assignable to type 'Atom<unknown>'.
src/store/assets/fungible-tokens.ts(20,3): error TS2345: Argument of type '([contractId]: [string, string]) => Atom<Promise<FungibleTokenMetadata | { error: string; } | undefined>>' is not assignable to parameter of type '(param: [string, string]) => Atom<FungibleTokenMetadata | { error: string; } | undefined>'.
  Type 'Atom<Promise<FungibleTokenMetadata | { error: string; } | undefined>>' is not assignable to type 'Atom<FungibleTokenMetadata | { error: string; } | undefined>'.
    Type 'Promise<FungibleTokenMetadata | { error: string; } | undefined>' is not assignable to type 'FungibleTokenMetadata | { error: string; } | undefined'.
      Property 'error' is missing in type 'Promise<FungibleTokenMetadata | { error: string; } | undefined>' but required in type '{ error: string; }'.
src/store/assets/fungible-tokens.ts(56,9): error TS2769: No overload matches this call.
  Overload 1 of 3, '(atom: Atom<unknown>): unknown', gave the following error.
    Argument of type 'FungibleTokenMetadata | { error: string; } | undefined' is not assignable to parameter of type 'Atom<unknown>'.
      Type 'undefined' is not assignable to type 'Atom<unknown>'.
  Overload 2 of 3, '(atom: Atom<Promise<unknown>>): unknown', gave the following error.
    Argument of type 'FungibleTokenMetadata | { error: string; } | undefined' is not assignable to parameter of type 'Atom<Promise<unknown>>'.
      Type 'undefined' is not assignable to type 'Atom<Promise<unknown>>'.
  Overload 3 of 3, '(atom: Atom<unknown>): unknown', gave the following error.
    Argument of type 'FungibleTokenMetadata | { error: string; } | undefined' is not assignable to parameter of type 'Atom<unknown>'.
      Type 'undefined' is not assignable to type 'Atom<unknown>'.
src/store/assets/fungible-tokens.ts(59,22): error TS2571: Object is of type 'unknown'.
src/store/assets/fungible-tokens.ts(60,61): error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'FtMeta | { error: string; }'.
  Type 'unknown' is not assignable to type '{ error: string; }'.
src/store/assets/fungible-tokens.ts(64,19): error TS2571: Object is of type 'unknown'.
src/store/assets/fungible-tokens.ts(65,15): error TS2571: Object is of type 'unknown'.
src/store/assets/fungible-tokens.ts(66,17): error TS2571: Object is of type 'unknown'.
src/store/network/networks.ts(12,30): error TS2554: Expected 3 arguments, but got 2.
src/store/network/networks.ts(20,37): error TS2554: Expected 3 arguments, but got 2.
src/store/onboarding/onboarding.ts(34,43): error TS2554: Expected 3 arguments, but got 2.
src/store/transactions/index.ts(57,11): error TS2339: Property 'transaction' does not exist on type '{ transaction: StacksTransaction; options: { senderKey: string; nonce: number; txData: any; }; } | { transaction: undefined; options: {}; } | undefined'.
src/store/transactions/index.ts(57,24): error TS2339: Property 'options' does not exist on type '{ transaction: StacksTransaction; options: { senderKey: string; nonce: number; txData: any; }; } | { transaction: undefined; options: {}; } | undefined'.
src/store/transactions/post-conditions.hooks.ts(16,31): error TS2571: Object is of type 'unknown'.
src/store/transactions/requests.hooks.ts(49,45): error TS2571: Object is of type 'unknown'.
src/store/transactions/transaction.hooks.ts(135,47): error TS2571: Object is of type 'unknown'.
src/store/ui/ui.ts(7,18): error TS2554: Expected 3 arguments, but got 2.
src/store/wallet/wallet.ts(16,30): error TS2554: Expected 3 arguments, but got 2.
src/store/wallet/wallet.ts(20,60): error TS2769: No overload matches this call.
  Overload 1 of 5, '(read: Read<WalletConfig | null>): Atom<WalletConfig | null>', gave the following error.
    Argument of type '(get: Getter) => Promise<WalletConfig | null>' is not assignable to parameter of type 'Read<WalletConfig | null>'.
      Property 'accounts' is missing in type 'Promise<WalletConfig | null>' but required in type 'WalletConfig'.
  Overload 2 of 5, '(initialValue: WalletConfig | null): Atom<WalletConfig | null> & { write: Write<SetStateAction<WalletConfig | null>, void>; onMount?: OnMount<...> | undefined; } & WithInitialValue<...>', gave the following error.
    Argument of type '(get: Getter) => Promise<WalletConfig | null>' is not assignable to parameter of type 'WalletConfig'.
      Property 'accounts' is missing in type '(get: Getter) => Promise<WalletConfig | null>' but required in type 'WalletConfig'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>


Details on breaking changes here https://github.com/pmndrs/jotai/releases/tag/v1.4.0

--- a/package.json
+++ b/package.json
@@ -116,7 +116,7 @@
     "fast-deep-equal": "3.1.3",
     "formik": "2.2.9",
     "http-server": "14.0.0",
-    "jotai": "1.3.2",
+    "jotai": "1.5.3",
     "jotai-query-toolkit": "0.1.7",
     "jsontokens": "3.0.0",
     "mdi-react": "7.5.0",

--- a/src/app/common/hooks/use-submit-stx-transaction.ts
+++ b/src/app/common/hooks/use-submit-stx-transaction.ts
@@ -63,12 +63,12 @@ export function useSubmitTransactionCallback({ loadingKey }: UseSubmitTransactio
         const nonce = !replaceByFee && Number(transaction.auth.spendingCondition?.nonce);
         try {
           const response = await broadcastTransaction(transaction, stacksNetwork);
-          if (typeof response !== 'string') {
+          if (response.error) {
             if (response.reason) toast.error(getErrorMessage(response.reason));
             onClose();
             setIsIdle();
           } else {
-            const txid = `0x${response}`;
+            const txid = `0x${response.txid}`;
             if (!externalTxid.includes(txid)) {
               await setLocalTxs({
                 rawTx: transaction.serialize().toString('hex'),

--- a/src/app/store/accounts/account-activity.ts
+++ b/src/app/store/accounts/account-activity.ts
@@ -18,7 +18,7 @@ type LocalTx = Record<
     timestamp: string;
   }
 >;
-const currentAccountLocallySubmittedTxsRootState = atomFamily<[string, string], LocalTx, LocalTx>(
+const currentAccountLocallySubmittedTxsRootState = atomFamily(
   ([_address, _network]) =>
     atomWithStorage<LocalTx>(makeLocalDataKey([_address, _network, 'LOCAL_TXS']), {}),
   deepEqual

--- a/src/app/store/accounts/api.ts
+++ b/src/app/store/accounts/api.ts
@@ -2,11 +2,7 @@ import { atomFamilyWithInfiniteQuery, atomFamilyWithQuery } from 'jotai-query-to
 import { QueryRefreshRates } from '@shared/constants';
 import { apiClientAnchoredState, apiClientState } from '@app/store/common/api-clients';
 
-import type {
-  AccountBalanceResponseBigNumber,
-  AddressBalanceResponse,
-  AccountBalanceStxKeys,
-} from '@shared/models/account-types';
+import type { AddressBalanceResponse, AccountBalanceStxKeys } from '@shared/models/account-types';
 
 import BigNumber from 'bignumber.js';
 import { atomFamily } from 'jotai/utils';
@@ -68,10 +64,7 @@ const accountBalancesAnchoredClient = atomFamilyWithQuery<
   }
 );
 
-export const accountBalancesAnchoredBigNumber = atomFamily<
-  PrincipalWithNetworkUrl,
-  AccountBalanceResponseBigNumber
->(
+export const accountBalancesAnchoredBigNumber = atomFamily(
   ({ principal, networkUrl }) =>
     atom(get => {
       const balances = get(accountBalancesAnchoredClient({ principal, networkUrl }));

--- a/src/app/store/accounts/index.ts
+++ b/src/app/store/accounts/index.ts
@@ -112,7 +112,7 @@ export const currentAccountStxAddressState = atom<string | undefined>(
   get => get(currentAccountState)?.address
 );
 
-const accountAvailableAnchoredStxBalanceState = atomFamily<string, BigNumber | undefined>(
+const accountAvailableAnchoredStxBalanceState = atomFamily(
   principal =>
     atom(get => {
       const networkUrl = get(currentNetworkState).url;

--- a/src/app/store/accounts/nonce.ts
+++ b/src/app/store/accounts/nonce.ts
@@ -12,7 +12,7 @@ import { currentNetworkState } from '@app/store/network/networks';
 import { makeLocalDataKey } from '@app/common/store-utils';
 import { currentAccountLocallySubmittedLatestNonceState } from '@app/store/accounts/account-activity';
 
-export const localNonceState = atomFamily<[principal: string, networkUrl: string], number, number>(
+export const localNonceState = atomFamily(
   params => atomWithStorage(makeLocalDataKey(['LOCAL_NONCE_STATE', params]), 0),
   deepEqual
 );

--- a/src/app/store/transactions/fees.hooks.ts
+++ b/src/app/store/transactions/fees.hooks.ts
@@ -24,7 +24,7 @@ export const useReplaceByFeeSubmitCallBack = () => {
   return useAtomCallback<void, { fee: number; nonce: number }>(
     useCallback(
       async get => {
-        const unsignedTx = await get(rawDeserializedTxState, true);
+        const unsignedTx = await get(rawDeserializedTxState, { unstable_promise: true });
         if (!unsignedTx) return;
         const signedTx = signTx(unsignedTx);
         if (!signedTx) return;

--- a/src/app/store/transactions/index.ts
+++ b/src/app/store/transactions/index.ts
@@ -72,7 +72,7 @@ export const unsignedStacksTransactionBaseState = atom(get => {
 export const unsignedStacksTransactionState = atom(get => {
   const { transaction, options } = get(unsignedStacksTransactionBaseState);
   if (!transaction) return;
-  return generateUnsignedTransaction({ ...options });
+  return generateUnsignedTransaction({ ...options } as any);
 });
 
 export function prepareTxDetailsForBroadcast(tx: StacksTransaction) {

--- a/src/app/store/transactions/transaction.hooks.ts
+++ b/src/app/store/transactions/transaction.hooks.ts
@@ -107,7 +107,7 @@ export function useTransactionBroadcast() {
             network: currentNetworkState,
             unsignedStacksTransaction: unsignedStacksTransactionState,
           }),
-          true
+          { unstable_promise: true }
         );
 
         if (!account || !requestToken || !unsignedStacksTransaction) {

--- a/src/app/store/ui/ui.ts
+++ b/src/app/store/ui/ui.ts
@@ -3,7 +3,7 @@ import { atom } from 'jotai';
 import { makeLocalDataKey } from '@app/common/store-utils';
 import { AccountStep } from './ui.models';
 
-export const tabState = atomFamily<string, number, number>(param => {
+export const tabState = atomFamily(param => {
   const anAtom = atomWithStorage<number>(makeLocalDataKey(['HOME_TABS', param]), 0);
   anAtom.debugLabel = `TABS__${param}`;
   return anAtom;
@@ -11,7 +11,7 @@ export const tabState = atomFamily<string, number, number>(param => {
 
 type LoadingState = 'idle' | 'loading';
 
-export const loadingState = atomFamily<string, LoadingState, LoadingState>(_param => {
+export const loadingState = atomFamily(_param => {
   const anAtom = atom<LoadingState>('idle');
   anAtom.debugLabel = `loading-atom/${_param}`;
   return anAtom;

--- a/src/app/store/wallet/wallet.ts
+++ b/src/app/store/wallet/wallet.ts
@@ -1,17 +1,12 @@
 import { atom } from 'jotai';
-import {
-  Wallet,
-  WalletConfig,
-  fetchWalletConfig,
-  createWalletGaiaConfig,
-} from '@stacks/wallet-sdk';
+import { Wallet, fetchWalletConfig, createWalletGaiaConfig } from '@stacks/wallet-sdk';
 import { gaiaUrl } from '@shared/constants';
 
 export const secretKeyState = atom<Uint8Array | undefined>(undefined);
 export const hasSetPasswordState = atom<boolean>(false);
 export const walletState = atom<Wallet | undefined>(undefined);
 export const encryptedSecretKeyState = atom<string | undefined>(undefined);
-export const walletConfigState = atom<WalletConfig | null>(async get => {
+export const walletConfigState = atom(async get => {
   const wallet = get(walletState);
   if (!wallet) return null;
 

